src/settings.rs

```diff
@@ -245,16 +150,12 @@ mod tests {
     }
 
     #[test]
-    fn name_regex_validation_should_not_allow_name_starting_with_numbers_test() -> Result<(), ()> {
+    fn name_regex_validation_should_not_allow_name_starting_with_numbers_test() {
         let environment_variable_name_regex = Regex::new(ENVIRONMENT_VARIABLE_NAME_REGEX).unwrap();
-        assert_eq!(environment_variable_name_regex.is_match("1envvar"), false);
-        assert_eq!(environment_variable_name_regex.is_match("2envvar2"), false);
-        assert_eq!(environment_variable_name_regex.is_match("3env_var3"), false);
-        assert_eq!(
-            environment_variable_name_regex.is_match("4_env_var4"),
-            false
-        );
-        Ok(())
+        assert!(!environment_variable_name_regex.is_match("1envvar"));
+        assert!(!environment_variable_name_regex.is_match("2envvar2"));
+        assert!(!environment_variable_name_regex.is_match("3env_var3"));
+        assert!(!environment_variable_name_regex.is_match("4_env_var4"));
     }
```

Comment:
I would replace these two tests with something written with rstest. I would make sure some invalid variables are spotted when doing settings validation. I would NOT write the tests against the regexp